lib/services/local/users.go

```diff
@@ -33,10 +34,17 @@ import (
 	"github.com/gravitational/teleport/lib/defaults"
 	"github.com/gravitational/teleport/lib/services"
 	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/crypto/bcrypt"
 )
 
+// GlobalSessionDataMaxEntries represents the maximum number of in-flight
+// global WebAuthn challenges for a given scope.
+// Attempting to write more instances than the max limit causes an error.
+// The limit is enforced separately by Auth Server instances.
+var GlobalSessionDataMaxEntries = 5000 // arbitrary
```

Comment:
From what I read below, we can allow double of this amount in the worst case?

For example

- in the first period of the limiter,  4999 upserts happened in the last second so no deletes for these in first period.
- then limiter resets to 0
- in the second period, we received another new 5000 upserts, then followed by 4999 deletes originated from the IDs from first period. 
- at this point, the count is 1, so we can allow another 4999 requests in the second period.

I think it's totally fine if this happens in extreme cases. just want to confirm if my understanding of the logic is correct.

Thanks.

Reply:
Yes, that sounds about right. Barring Write/Delete DB throughput, there are various scenarios where we can get borderline infinite writes/second and that's fine - it means we are consuming challenges as fast as we make new ones, a sign of healthy interactions.

The "degenerate" scenario is having a much higher amount of created than deleted challenges, as it could mean a malicious interaction.